numba/targets/arraymath.py

```diff
@@ -1294,6 +1294,26 @@ def np_vander_seq_impl(x, N=None, increasing=False):
     elif isinstance(x, (types.Tuple, types.Sequence)):
         return np_vander_seq_impl
 
+@overload(np.roll)
+def np_roll(a, shift):
+
+    if not isinstance(shift, (types.Integer, types.Boolean)):
+        raise TypingError('shift must be an integer')
+
+    def np_roll_impl(a, shift):
+        arr = _asarray(a)
```

Comment:
Bah - it's a 0D array regardless of shift.

I thought this would be a one-liner but my tests fail depending on what order I run them in, which I appreciate is somewhat suboptimal :)

Here's a simple repro:

```python
from numba import njit
from numba.extending import overload, types
import numpy as np


def foo(a, shift):
    pass


@overload(foo)
def foo_impl(a, shift):

    def foo_impl_0d(a, shift):
        return np.array(a)

    if isinstance(a, (types.Boolean, types.Number)):
        return foo_impl_0d

@njit
def foo_quickly(a, shift):
    return foo(a, shift)


def check_stuff(a_values):
    for idx, a in enumerate(a_values):
        got = foo_quickly(a, 1).item(0)
        assert got is a, 'got {0} but expected {1}'.format(got, a)


if __name__ == '__main__':

    # this passes
    check_stuff([False, True])
    check_stuff([0, 1])

    # comment out the above and un-comment the below: fails
    # check_stuff([0, 1])
    # check_stuff([False, True])
```

Per the comments, the assertions pass if I run the tests in one order, but fail if I switch the order.

If you inspect the signatures of `foo_quickly` then the former case ends up with [(bool, int64), (int64, int64)] whilst the latter only has [(int64, int64)]).

Have I made some festive mishap?

Reply:
I think this needs #3612

Reply:
I simply swapped the test order - #3612 would mean I didn't have to, but I figured just as easy to order the tests.